benchmarking/blackbox_repository/blackbox_surrogate.py

```diff
@@ -58,22 +80,27 @@ def __init__(
         # todo other types of assert with configuration_space, objective_names, ...
         self.X = X
         self.y = y
-        self.surrogate = surrogate
-        self.fit_surrogate(surrogate)
+        if surrogate is None:
+            surrogate = 'nn-1'
+        self.surrogate = blackbox_surrogate_factory(surrogate)
```

Comment:
This will break user code who use a surrogate not covered by the factory (I currently have a few cases that are not supported by the factory). It will also prevent the use from custom surrogate without touching the factory. 

I would propose to rather save only surrogate options if the size is an issue (I can make that change). It seems that the original issue comes from a sklearn version also, we can have a look together to fix it in the meantime.

Reply:
Hello David, this does not break previous code. If surrogate is not a string, it does the same as before. So you can keep doing what you did before. It just may fail.

To be honest (and without disrespect), but I feel I am wasting time with these serialization of complex objects issues. I told you before that I am sceptical for exactly this reason. It is also not good practice to configre options by complex code and just serialize it, and hope for the best. It just calls for trouble every time there is any change anywhere.

Reply:
And the solution with clone is doing what it's supposed to do, without having to dig into which estimators have what parameters.

Reply:
To be specific: If surrogate is not a string, the factory returns surrogate (it does not fail). So, if you pass your favourite sklearn estimator, or anything obeying its API, you are fine.